executors/http/src/main/java/net/opentsdb/utils/DefaultSharedHttpClient.java

```diff
@@ -0,0 +1,286 @@
+// This file is part of OpenTSDB.
+// Copyright (C) 2018  The OpenTSDB Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package net.opentsdb.utils;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+import javax.net.ssl.SSLContext;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.ParseException;
+import org.apache.http.client.entity.DeflateDecompressingEntity;
+import org.apache.http.client.entity.GzipDecompressingEntity;
+import org.apache.http.impl.nio.client.CloseableHttpAsyncClient;
+import org.apache.http.impl.nio.client.HttpAsyncClients;
+import org.apache.http.impl.nio.reactor.IOReactorConfig;
+import org.apache.http.util.EntityUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.base.Strings;
+import com.stumbleupon.async.Deferred;
+
+import io.netty.util.Timeout;
+import io.netty.util.TimerTask;
+import net.opentsdb.core.BaseTSDBPlugin;
+import net.opentsdb.core.TSDB;
+import net.opentsdb.exceptions.RemoteQueryExecutionException;
+import net.opentsdb.utils.RefreshingSSLContext.RefreshCallback;
+import net.opentsdb.utils.RefreshingSSLContext.SourceType;
+
+/**
+ * A shared asynchronous HTTP client for making remote calls to various
+ * services.
+ * 
+ * @since 3.0
+ * @author clarsen
+ *
+ */
+public class DefaultSharedHttpClient extends BaseTSDBPlugin implements RefreshCallback {
```

Comment:
What do you mean by Shared here?

Reply:
Looks like just a name. Nothing special about Sharing here.

Reply:
As it multiple sources (e.g. multi-colo) can use the same client pool instead of instantiating one per node.